pkg/block/fetcher.go

```diff
@@ -574,4 +596,37 @@ func (f *ConsistencyDelayMetaFilter) Filter(metas map[ulid.ULID]*metadata.Meta,
 			delete(metas, id)
 		}
 	}
+
+	return nil
+}
+
+// DeletionMarkMetaFilter is a MetaFetcher that filters blocks that are marked for deletion (deleteDelay/2) duration before current time.
+type DeletionMarkMetaFilter struct {
```

Comment:
```suggestion
type IgnoreDeletionMarkFilter struct {
```
Let's be explicit. Why meta was needed?